src/components/media-loader.js

```diff
@@ -396,6 +396,11 @@ AFRAME.registerComponent("media-loader", {
       // we don't think we can infer it from the extension, we need to make a HEAD request to find it out
       contentType = contentType || guessContentType(canonicalUrl) || (await fetchContentType(accessibleUrl));
 
+      // TODO we should probably just never return "application/octet-stream" as expectedContentType, since its not really useful
+      if (contentType === "application/octet-stream") {
+        contentType = guessContentType(canonicalUrl) || contentType;
```

Comment:
Not sure what the expectation is for getting to this line. If we got `"application/octet-stream"` and then `guessContentType(canonicalUrl)` failed, should we `await fetchContentType(accessibleUrl)` like line https://github.com/mozilla/hubs/pull/2448/files#diff-50f3f477609a941f9eafc236322b8160R397 above?

Reply:
As noted in the TODO, best way to handle this would be to have the server just not return `application/octet-stream`, until we do that I just wanted to make the least risky change possible, and didn't want to accidentally trigger a second head request nor muck with the logic too much, so just falling back to guessing content type for known filenames as a last resort.